src/ui/EditorNS/editor.cpp

```diff
@@ -206,10 +236,27 @@ namespace EditorNS
 
     QList<QMap<QString, QString>> Editor::languages()
     {
+        // TODO: Cache this data.
+        QJSEngine engine;
+        QFileInfo fileInfo(Notepadqq::editorPath());
+        QString fileName = fileInfo.absolutePath() + "/classes/Languages.js";
```

Comment:
Why not do it in the old way just the first time, and then cache it for all the following times?
It would have the advantage to remove the dependency on qml and to avoid introducing assumptions about the javascript code (which should remain an abstraction).